src/mc_tasks/lipm_stabilizer/StabilizerTask.cpp

```diff
@@ -1128,6 +1151,337 @@ void StabilizerTask::distributeWrench(const sva::ForceVecd & desiredWrench)
+      rightContact.halfWidth();
+
+  Eigen::Vector2d t_lankle_zmp = zmp_ref[0] - lankle.segment(0, 2);
+  const double lankle_rankle = t_lankle_rankle.norm();
+  double d_proj = t_lankle_zmp.dot(t_lankle_rankle.normalized());
+  // The vertical forces are splitted using the ratio obtained between the reference zmp pose and the contact pose;
+  double ratio = clamp(d_proj / lankle_rankle, c_.safetyThresholds.MIN_DS_PRESSURE / fz_tot,
+                       1 - (c_.safetyThresholds.MIN_DS_PRESSURE / fz_tot));
+
+  desiredFzLeft_ = (1 - ratio) * fz_tot;
+  desiredFzRight_ = ratio * fz_tot;
+
+  // We modeled the vertical forces to also follow a 1st order behavior between the reference and the state
+  targetForceLeft.z() = desiredFzLeft_ - measuredFzLeft_delayed * exp(-c_.lambda_CoP_Fz.z() * (delta - t_delay));
+  targetForceLeft.z() /= (1 - exp(-c_.lambda_CoP_Fz.z() * (delta - t_delay)));
+
+  targetForceRight.z() = desiredFzRight_ - measuredFzRight_delayed * exp(-c_.lambda_CoP_Fz.z() * (delta - t_delay));
+  targetForceRight.z() /= (1 - exp(-c_.lambda_CoP_Fz.z() * (delta - t_delay)));
+
+  const double ratio0 = ratio;
+
+  for(Eigen::Index i = 0; i < nbReferences; i++)
+  {
+
+    if(i != 0)
+    {
+
+      t_lankle_zmp = zmp_ref[i] - lankle.segment(0, 2);
+      d_proj = t_lankle_zmp.dot(t_lankle_rankle.normalized());
+      // ratio = 1 : fz on rightfoot, 0 on leftFoot
+      ratio = clamp(d_proj / lankle_rankle, c_.safetyThresholds.MIN_DS_PRESSURE / fz_tot,
+                    1 - (c_.safetyThresholds.MIN_DS_PRESSURE / fz_tot));
+    }
+
+    // Acop convert the CoP reference into the modeled CoP
+    // x be vector cotaining the CoP reference for one foot
+    // Acop * x + cop_0 * e^(-lambda t_i) = cop i in foot frame (left/right)
+    Eigen::MatrixXd Acop = Eigen::MatrixXd::Zero(2, 2 * nbReferences);
+    double t = static_cast<double>(i) * delta;
+    Eigen::Matrix2d exp_mat;
+    exp_mat << exp(-c_.lambda_CoP_Fz.x() * (t + delta - (i == 0 ? t_delay : c_.delayCoP))), 0, 0,
+        exp(-c_.lambda_CoP_Fz.y() * (t + delta - (i == 0 ? t_delay : c_.delayCoP)));
+    for(Eigen::Index k = 0; k <= i; k++)
+    {
+      if(k == i)
+      {
+        Acop(0, 2 * k) = (1 - exp(-c_.lambda_CoP_Fz.x() * (delta - (i == 0 ? t_delay : c_.delayCoP))))
+                         * exp(-c_.lambda_CoP_Fz.x() * t);
+        Acop(1, 2 * k + 1) = (1 - exp(-c_.lambda_CoP_Fz.y() * (delta - (i == 0 ? t_delay : c_.delayCoP))))
+                             * exp(-c_.lambda_CoP_Fz.y() * t);
+      }
+      else
+      {
+        Acop(0, 2 * k) = (1 - exp(-c_.lambda_CoP_Fz.x() * delta)) * exp(-c_.lambda_CoP_Fz.x() * t);
+        Acop(1, 2 * k + 1) = (1 - exp(-c_.lambda_CoP_Fz.y() * delta)) * exp(-c_.lambda_CoP_Fz.y() * t);
+      }
+      t -= delta;
+    }
+
+    // The task regulate zmp_i = (cop_l * f_z_l + cop_r * f_z_r)/f_z in world frame
+    Mcop.block(2 * i, 0, 2, 2 * nbReferences) = X_0_lc.inv().rotation().block(0, 0, 2, 2) * (1 - ratio) * Acop;
```

Comment:
Avoid copying known zero to zero
```suggestion
    auto Acop_view = Acop.block(0, 0, 2, 2 * (i + 1));
    Mcop.block(2 * i, 0, 2, Acop_view.cols()) = X_0_lc.inv().rotation().block(0, 0, 2, 2) * (1 - ratio) * Acop_view;
```
(note: for once, auto is the right choice here)